cmd/libsnap-confine-private/snap-test.c

```diff
@@ -118,13 +118,13 @@ static void test_sc_is_hook_security_tag(void)
 	g_assert_true(sc_is_hook_security_tag
 		      ("snap.foo_instance.hook.bar-baz"));
 	g_assert_true(sc_is_hook_security_tag("snap.foo_bar.hook.bar-baz"));
+	g_assert_true(sc_is_hook_security_tag("snap.foo_bar.hook.f00"));
 
 	// Now, test the names we know are not valid hook security tags
 	g_assert_false(sc_is_hook_security_tag("snap.foo_instance.bar-baz"));
 	g_assert_false(sc_is_hook_security_tag("snap.name.app!hook.foo"));
 	g_assert_false(sc_is_hook_security_tag("snap.name.app.hook!foo"));
 	g_assert_false(sc_is_hook_security_tag("snap.name.app.hook.-foo"));
-	g_assert_false(sc_is_hook_security_tag("snap.name.app.hook.f00"));
```

Comment:
this one can be left in the test cases as invalid, though reading all these invalid cases we should add some more invalid ones that are _closer_ to being correct such as

* `snap.foo_bar.hook.!foo`
* `snap.foo_bar.hook.-foo`
* `snap.foo_bar.hook!foo`
* `snap.foo_bar.!foo`

Reply:
also add `snap.foo_bar.hook.0abcd` and `snap.foo.hook.abc--` to the list of invalid entries

Reply:
Added, thanks for suggestions.